### PR TITLE
DEX-484: add content_guid to Annotation model

### PR DESCRIPTION
**Pull request by naknomum**
## Pull Request Overview

- Add `content_guid` to Annotation model
- Set `content_guid` from `guid` value in annotation data returned from **detection** results
- Primary key `guid` now is uuidv4 random
- _Bonus (not in ticket):_ add `viewpoint` property to Annotation (and set from detection results); added to schema

---

**Database Updates**

- `content_guid` added (nullable, following lead from Asset model)
- `viewpoint` added (not nullable)
- _Note:_ db migration first sets `viewpoint` nullable=True in case Annotations exist in db; then it sets viewpoint=unknown; then sets nullable=False

### CODE REVIEW

File: app/modules/asset_groups/models.py [hunk 1]
```diff
@@ -461,19 +461,22 @@ def detected(self, job_id, response):
...
                 annot_data = results[annot_id]
-                annot_uuid = annot_data.get('uuid', {}).get('__UUID__')
+                content_guid = annot_data.get('uuid', {}).get('__UUID__')
                 ia_class = annot_data.get('class', None)
-                if not annot_uuid or not ia_class:
+                viewpoint = annot_data.get('viewpoint', None)
+                if not viewpoint or not ia_class:
                     raise HoustonException(
-                        log, 'Need a uuid and a class in each of the results'
+                        log, 'Need a viewpoint and a class in each of the results'
```
Comment (hwindsor) [inline app/modules/asset_groups/models.py:469]: This is an I don't know question buut does this mean that the content guid is optioonal? If so how/why?

File: tests/modules/audit_logs/resources/test_audit_log.py [hunk 2]
```diff
@@ -173,7 +173,7 @@ def test_most_ia_pipeline_audit_log(
 
     assert expected_sighting in sighting_audit_items.json
     assert expected_encounter in encounter_audit_items.json
-    assert expected_annotation in annotation_audit_items.json
+    assert expected_annotation not in annotation_audit_items.json
```
Comment (hwindsor) [inline tests/modules/audit_logs/resources/test_audit_log.py:176]: Yes, if we can no longer know what the annotation guid, this line plus the two things it's checking are not needed anymore.

Comment (naknomum) [reply tests/modules/audit_logs/resources/test_audit_log.py:176]: ok, i will go ahead and just remove the annot audit_log test then.  thanks.

File: app/modules/asset_groups/models.py (hunk 1 above)
Comment (naknomum) [reply app/modules/asset_groups/models.py:469]: yes - or so i claim.  as to why; well, i am just basing this on the [analog in Asset model](https://github.com/WildMeOrg/houston/blob/main/app/modules/assets/models.py#L44).  in the annot case, when it is created from detection, we _should_ always have this value to set.  when we (eventually) manually create annots in the future, we would have to compute it.

perhaps the real question (outside the scope of this pr?) is: what is it used for?  in wbia, it is my understanding it is greatly important to know when you are actually talking about "the same annotation" from a visual/pixel/content point of view.  do we ever care about that in houston or from a user standpoint?  not really right now anyway.   so i think optional is fine.  (it seems like we currently _do not_ set the one on asset at this point, fwiw.)

**Review by hwindsor — APPROVED** [review]
LGTM

**Pull request merged**